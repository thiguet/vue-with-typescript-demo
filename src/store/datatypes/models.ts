interface Unique {
    id?: string;
}

export enum Measures {
    UN = 'UN',
    L = 'L',
    KG = 'KG',
    G = 'G',
}

export interface Product extends Unique {
    name: string;
    measure: Measures;
    qtd: number;
    minQtd: number;
    image?: string;
}

export interface User extends Unique {
    name: string;
    email: string;
}

export interface ReportItem {
    text: string;
    value: number;
}

export type ReportData = ReportItem[];

export interface QuantityReportData {
    productName: string;
    productQuantity: number;
}

export type QuantityReport = QuantityReportData[];
export interface MeasuresReportData {
    measureName: string;
    measureQuantity: number;
}

export type MeasuresReport = MeasuresReportData[];

export enum VuexAppModules {
    products = 'products',
    login = 'login',
    alert = 'alert',
    reports = 'reports',
}
