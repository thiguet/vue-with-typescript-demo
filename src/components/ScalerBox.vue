<template>
    <div class="box-container">
        <div class="box">
            <slot></slot>
        </div>
        <div class="info">
            <p>
                {{ boxTitle }}
            </p>
        </div>
    </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component, Prop } from 'vue-property-decorator';

@Component
export default class ScalerBox extends Vue {
    @Prop() private readonly boxTitle!: string;
}
</script>

<style>
.box-container {
    margin-top: 20px;
    padding: 20px;
    cursor: pointer;
    flex-flow: column;
    width: auto;
}

.box-container:hover .box > *,
.box-container:focus .box > * {
    transform: scale(1.5);
}

.box {
    width: auto;
    display: flex;
    margin-right: 0;
    filter: none;
    transition: all 0.3s ease;
}

.info,
.box {
    justify-content: center;
}

@media screen and (max-width: 600px) {
    .box-container {
        padding: 0;
    }
}
</style>
