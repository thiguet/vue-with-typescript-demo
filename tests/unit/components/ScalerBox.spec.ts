import { shallowMount, createLocalVue, Wrapper } from '@vue/test-utils';

import Vuex from 'vuex';

import ScalerBox from '@/components/ScalerBox.vue';
import faker from 'faker';
import { ScalerBoxProps } from './models';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Scaler Box Props', () => {
    const slotContent = '<div>Main Content</div>';
    let props: ScalerBoxProps;

    const build = () => {
        const componentBindings = {
            propsData: { ...props },
            slots: {
                default: slotContent,
            },
        };

        const wrapper: Wrapper<ScalerBox> = shallowMount(
            ScalerBox,
            componentBindings,
        );

        return {
            wrapper,
            boxContainer: () => wrapper.find('.box-container'),
            box: () => wrapper.find('.box'),
            info: () => wrapper.find('.info'),
        };
    };

    beforeEach(() => {
        props = {
            boxTitle: faker.random.words(),
        };
    });

    it('renders component', () => {
        props.boxTitle = 'teste';
        const { wrapper } = build();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders child components', () => {
        const { box, boxContainer, info } = build();
        expect(box().exists()).toBe(true);
        expect(boxContainer().exists()).toBe(true);
        expect(info().exists()).toBe(true);
    });

    it('renders slot data', () => {
        const { box } = build();
        expect(box().html()).toContain(slotContent);
    });
    it('sets prop title to the info section', () => {
        const { wrapper, info } = build();
        expect(info().text()).toEqual(wrapper.props().boxTitle);
    });
});
