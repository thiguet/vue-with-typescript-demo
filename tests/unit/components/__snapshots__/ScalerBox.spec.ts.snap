// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scaler Box Props renders component 1`] = `
<div class="box-container">
  <div class="box">
    <div>Main Content</div>
  </div>
  <div class="info">
    <p>
      teste
    </p>
  </div>
</div>
`;
